src/utils/KML.js

```diff
@@ -306,6 +306,12 @@ const writeFeatures = (layer, featureProjection) => {
       }
     }
 
+    if (newStyle.fill && newStyle.fill.getColor() instanceof CanvasPattern) {
```

Comment:
You should exclude also CanvasGradient

Reply:
and test if f.get('fillPattern') exists . maybe just a :

``` 
if (newStyle.fill &&  f.get('fillPattern'))
``` 

is enough

Reply:
@oterral Did you mean something like this?
`if (
      (newStyle.fill && newStyle.fill.getColor() instanceof CanvasPattern) ||
      newStyle.fill.getColor() instanceof CanvasGradient
    ) {
      if (f.get('fillPattern')) {
        clone.set('fillPattern', f.get('fillPattern'));
        newStyle.fill = null;
      }
    }`

Reply:
no just :
``` 
if (newStyle.fill &&  f.get('fillPattern')) {
   clone.set('fillPattern', f.get('fillPattern'));
   newStyle.fill = null;
}
```